cpp.mk

```diff
@@ -44,9 +44,8 @@ $1_DEPS := $$(patsubst %.cpp,$${BUILD_DIR}/$1/%.d,$${$1_SOURCES})
 # header files)
 $1_OBJECT_DIR := $${BUILD_DIR}/$1
 
-# Depend on the depends file so we'll get rebuilt if it's missing
-# (which will have the side effect of re-producing the depends file)
-$${$1_OBJS} : $${$1_OBJECT_DIR}/%.o : %.cpp $${BUILD_DIR}/$1/%.d
+# Object files are produced by compiling source files
+$${$1_OBJS} : $${$1_OBJECT_DIR}/%.o : %.cpp
```

Comment:
Either I misunderstood the conversation or that doesn't work.  Can you spell out the plan for me?

Alternatively, [this page](http://make.mad-scientist.net/papers/advanced-auto-dependency-generation) which you pointed me at does in fact take the approach that we'd previously discarded as hacky:
- we'd considered an 'ugly' "touch the `.d` file after compiling" step
- that model code has a `POSTCOMPILE` step that does more or less the same thing.

So perhaps back to that plan?